RevenueCatUI/Templates/V2/Components/Stack/FlexHStack.swift

```diff
@@ -40,48 +40,82 @@ struct FlexHStack: View {
                 Spacer(minLength: 0)
                 ForEach(0..<componentViewModels.count, id: \.self) { index in
                     ComponentsView(componentViewModels: [self.componentViewModels[index]], onDismiss: self.onDismiss)
+                    if index < self.componentViewModels.count - 1 {
+                        if let spacing = self.spacing {
+                            Spacer(minLength: spacing)
+                        }
+                    }
                 }
                 Spacer(minLength: 0)
 
             case .end:
                 Spacer(minLength: 0)
                 ForEach(0..<componentViewModels.count, id: \.self) { index in
                     ComponentsView(componentViewModels: [self.componentViewModels[index]], onDismiss: self.onDismiss)
+                    if index < self.componentViewModels.count - 1 {
+                        if let spacing = self.spacing {
+                            Spacer(minLength: spacing)
+                        }
+                    }
                 }
 
             case .spaceBetween:
                 ForEach(0..<componentViewModels.count, id: \.self) { index in
                     ComponentsView(componentViewModels: [self.componentViewModels[index]], onDismiss: self.onDismiss)
                     if index < self.componentViewModels.count - 1 {
-                        Spacer(minLength: 0)
+                        if let spacing = self.spacing {
+                            Spacer(minLength: spacing)
+                        } else {
+                            Spacer(minLength: 0)
+                        }
                     }
                 }
 
             case .spaceAround:
                 ForEach(0..<componentViewModels.count, id: \.self) { index in
-                    Spacer(minLength: 0)
+                    if index == 0 {
+                        FlexSpacer(weight: 1)
+                    }
                     ComponentsView(componentViewModels: [self.componentViewModels[index]], onDismiss: self.onDismiss)
-                    Spacer(minLength: 0)
+                    if index < self.componentViewModels.count - 1 {
+                        if let spacing = self.spacing {
+                            Spacer(minLength: spacing)
+                        }
+                        FlexSpacer(weight: 2)
+                    } else {
+                        FlexSpacer(weight: 1)
+                    }
                 }
 
             case .spaceEvenly:
                 ForEach(0..<componentViewModels.count, id: \.self) { index in
-                    Spacer(minLength: 0)
+                    FlexSpacer(weight: 1)
                     ComponentsView(componentViewModels: [self.componentViewModels[index]], onDismiss: self.onDismiss)
+                    if index < self.componentViewModels.count - 1 {
+                        if let spacing = self.spacing {
+                            Spacer(minLength: spacing)
+                        }
```

Comment:
For Spacers that are supposed to take up exactly a set amount of space, we shouldn't use minLength. We should use `Spacer().frame(width: spacing, height: spacing)` Then we get the precise spacing